modules/c++/mem/include/mem/ScratchMemory.hpp

```diff
@@ -52,7 +56,83 @@ inline void ScratchMemory::put<sys::ubyte>(const std::string& key,
     }
     mSegments.insert(
             iterSeg,
-            std::make_pair(key, Segment(numElements, numBuffers, alignment)));
+            std::make_pair(key, Segment(numElements, numBuffers, alignment, segmentOffset)));
+
+    mKeyOrder.push_back(key);
+}
+
+inline void ScratchMemory::release(const std::string& key)
```

Comment:
Remove the inline keyword here. The compiler probably won't inline this.

Reply:
The compiler is giving me a "multiple definition" error when I do so, I think because the function is in both the .h and .hpp?

Reply:
Ahh, that is because this method is not templated. Can you move this to a cpp file?